api/src/main/java/io/smallrye/stork/api/Service.java

```diff
@@ -12,39 +15,110 @@
  */
 public class Service {
 
+    private final Lock instanceSelectionLock;
     private final Optional<LoadBalancer> loadBalancer;
     private final ServiceDiscovery serviceDiscovery;
     private final String serviceName;
     private final boolean secure;
 
-    public Service(String serviceName, Optional<LoadBalancer> loadBalancer, ServiceDiscovery serviceDiscovery, boolean secure) {
+    public Service(String serviceName, Optional<LoadBalancer> loadBalancer, ServiceDiscovery serviceDiscovery,
+            boolean secure, boolean threadSafeLoadBalancer) {
         this.loadBalancer = loadBalancer;
         this.serviceDiscovery = serviceDiscovery;
         this.serviceName = serviceName;
         this.secure = secure;
+        instanceSelectionLock = threadSafeLoadBalancer ? null : new ReentrantLock();
```

Comment:
```suggestion
       this.instanceSelectionLock = threadSafeLoadBalancer ? null : new ReentrantLock();
```

Reply:
I don't like those thises :D but you're right it looks better if all assignments have them than with one not having it